src/common/updating/updateState.cpp

```diff
@@ -34,19 +37,40 @@ json::JsonObject serialize(const UpdateState& state)
     json.SetNamedValue(L"state", json::value(static_cast<double>(state.state)));
     json.SetNamedValue(L"downloadedInstallerFilename", json::value(state.downloadedInstallerFilename));
 
+    json.SetNamedValue(L"updateStateFileVersion", json::value(CURRENT_VERSION.toWstring()));
+
     return json;
 }
 
+bool IsOldFileVersion(const std::wstring_view fileVersion)
+{
+    if (fileVersion == L"")
+    {
+        return true;
+    }
+
+    const auto parsedVer = VersionHelper::fromString(fileVersion);
+    return !parsedVer.has_value() || *parsedVer < CURRENT_VERSION;
```

Comment:
This can be problematic when doing a downgrade. Then `parsedVer` won't be lower than `CURRENT_VERSION`.

Would it not be better to use unequal: `*parsedVer <> CURRENT_VERSION`?

Reply:
Yes, that makes sense, thanks!